### PR TITLE
Documented user folder configuration.

### LINKED ISSUE #125: Support configuration of favorite folders
**Is your feature request related to a problem? Please describe.**
I need to manually navigate to my favorite folder(s) to load and save sf2 files.

**Describe the solution you'd like**
Configure a list of my favorite folders so I can open them directly from the file dialog

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by micheldebree**
Documented how to configure user folders in config.ini and user.default.ini

There were some hardcoded shortcuts in platform linux and macos sources that I removed and configured in config.ini

Resolves #125

### CODE REVIEW

**Comment by micheldebree** [comment]
Also added Disk.Startup.Folder to configure the startup folder for the disk browser.

File: SIDFactoryII/source/foundation/platform/sdl/platform_sdl_macos.cpp [hunk 1]
```diff
@@ -3,196 +3,186 @@
...
-				// Add the user's home folder
-				char *realHome = getpwuid(getuid())->pw_dir;
+		uint32_t buffer_size = PATH_MAX;
+		_NSGetExecutablePath(m_ApplicationPath, &buffer_size);
 
-				if(realHome != nullptr)
-				{
-						m_RealHome = std::string(realHome);
+		// Add the user's home folder
+		char* realHome = getpwuid(getuid())->pw_dir;
```
Comment (TheRealLaxity) [inline SIDFactoryII/source/foundation/platform/sdl/platform_sdl_macos.cpp:27]: Looks like the memory buffer isn't freed again. Can you verify how the memory of char* is handled. If it's not supposed to be user managed, is suggest making the pointer const.

Comment (TheRealLaxity) [reply SIDFactoryII/source/foundation/platform/sdl/platform_sdl_macos.cpp:27]: Ok. So checked the documentation. You do not own the memory returned from the call, and so it should be const and you're not supposed to free it.

**Review by TheRealLaxity — APPROVED** [review]
Just noticed a thing that could be const, otherwise great as far as K can tell.

**Pull request merged**